Added type hints for TF: rag model

debjit-bw opened the pull request:
Based on Issue https://github.com/huggingface/transformers/issues/16059

Type hints for the [TFRagModel](https://huggingface.co/docs/transformers/model_doc/rag) have been added.

@Rocketknight1 Could you please check the changes and merge if it's fine?

Thanks a lot.

debjit-bw committed:
Added type hints for TF: rag model

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_19284). All of your documentation changes will be reflected on that endpoint.

debjit-bw commented:
@Rocketknight1 can you take a look at this? I just added type hints and fixed a syntax error in one previous type. Still it shows it failed these 2 tests. I have no idea what these tests do, can you shed some light on this issue?

Rocketknight1 commented on src/transformers/models/rag/modeling_tf_rag.py:
```diff
@@ -538,22 +538,22 @@ def set_retriever(self, retriever: RagRetriever):
-        n_docs=None,
-        return_dict=None,
-        training=False,
+        input_ids: Optional[tf.Tensor] = None,
```
```suggestion
        input_ids: Optional[TFModelInputType] = None,
```

Rocketknight1 commented on src/transformers/models/rag/modeling_tf_rag.py:
```diff
@@ -828,25 +828,25 @@ def marginalize(self, seq_logits, doc_scores, n_docs=None):
-        reduce_loss=None,
-        return_dict=None,
-        training=False,
+        input_ids: Optional[tf.Tensor] = None,
```
```suggestion
        input_ids: Optional[TFModelInputType] = None,
```

Rocketknight1 commented on src/transformers/models/rag/modeling_tf_rag.py:
```diff
@@ -1425,27 +1425,27 @@ def question_encoder(self):
-        reduce_loss=None,
-        return_dict=None,
-        training=False,
+        input_ids: Optional[tf.Tensor] = None,
```
```suggestion
        input_ids: Optional[TFModelInputType] = None,
```

Rocketknight1 reviewed (approved):
Overall this looks good! Don't worry about the test - that seems totally unrelated. You could probably fix it by pulling upstream commits to your fork's `main` branch and then rebasing your PR branch onto the main branch followed by a force-push, but don't worry if you're not sure how to do that - I'm happy to merge even with those tests failing.

My one issue is that the first input type for TF models should be `TFModelInputType`. You'll need to import that type from `modeling_tf_utils.py`.

debjit-bw committed:
TFModelInputType added in place of TF.Tensor

debjit-bw commented:
Thanks a lot @Rocketknight1, I have made the suggested changes!

debjit-bw committed:
reformatting by black
